### ISSUE TITLE
NotImplementedError: The admin_set_user_password action has not been implemented

### ISSUE THREAD

**Issue by robjwilkins** [spulec/moto#3476]
Problem: using moto to stub the Cognito IDP API, I get an error: `NotImplementedError: The admin_set_user_password action has not been implemented`

I am using docker moto image: `motoserver:moto:latest`:
```
Digest: sha256:ef4e4172daf2801e63856687d6f5a99df8ad4b39d86b56b91f9581df88d2358d
Status: Downloaded newer image for motoserver/moto:latest
docker.io/motoserver/moto:latest
```

The stacktrace:
```
Error on request:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/werkzeug/serving.py", line 323, in run_wsgi
    execute(self.server.app)
  File "/usr/local/lib/python3.7/site-packages/werkzeug/serving.py", line 312, in execute
    application_iter = app(environ, start_response)
  File "/usr/local/lib/python3.7/site-packages/moto/server.py", line 167, in __call__
    return backend_app(environ, start_response)
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2464, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1867, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.7/site-packages/moto/core/utils.py", line 146, in __call__
    result = self.callback(request, request.url, {})
  File "/usr/local/lib/python3.7/site-packages/moto/core/responses.py", line 202, in dispatch
    return cls()._dispatch(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/moto/core/responses.py", line 312, in _dispatch
    return self.call_action()
  File "/usr/local/lib/python3.7/site-packages/moto/core/responses.py", line 410, in call_action
    "The {0} action has not been implemented".format(action)
NotImplementedError: The admin_set_user_password action has not been implemented
```
I believe this feature should be available in the latest release? https://github.com/spulec/moto/pull/3328

The code which calls the moto API is using the java AWS SDK:
```
    var setUserPasswordRequest = AdminSetUserPasswordRequest.builder()
      .username(username)
      .password(password)
      .permanent(true)
      .userPoolId(userPoolId)
      .build();
    cognitoClient.adminSetUserPassword(setUserPasswordRequest);
```

Other calls (to create user pools, and create users) do work successfully. This call fails with the following stacktrace:
```
com.fasterxml.jackson.core.JsonParseException: Unexpected character ('<' (code 60)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (ByteArrayInputStream); line: 1, column: 2]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1851)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:707)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportUnexpectedChar(ParserMinimalBase.java:632)
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._handleUnexpectedValue(UTF8StreamJsonParser.java:2686)
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._nextTokenNotInObject(UTF8StreamJsonParser.java:865)
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser.nextToken(UTF8StreamJsonParser.java:757)
	at software.amazon.awssdk.protocols.json.internal.dom.JsonDomParser.parse(JsonDomParser.java:41)
	at software.amazon.awssdk.protocols.json.JsonContent.parseJsonContent(JsonContent.java:76)
	at software.amazon.awssdk.protocols.json.JsonContent.<init>(JsonContent.java:49)
	at software.amazon.awssdk.protocols.json.JsonContent.createJsonContent(JsonContent.java:67)
```

As far as I can tell the only issues regarding this topic are closed, and the branches merged.

thanks for your help.

**Comment by robjwilkins** [comment]
Interesting - I just cloaned the repo, and built the docker image myself, and tagged it. When I use this image in the test the error does not happen.

I therefore guess that the image published on docker hub does not include the latest changes in the python code?

**Comment by g-dx** [comment]
#3357

**Comment by robjwilkins** [comment]
I've pushed a working docker image for the moto project to a public docker hub repo as a work-around: https://hub.docker.com/r/robinwaracle/motoserver

**Comment by bblommers** [comment]
Thanks for the providing the workaround @robjwilkins.

I'll close this, as it is a duplicate of #3357 in the end.

**Comment by dwevans1994** [comment]
Using moto 1.3.16 and getting this:
- admin_create_user works fine, then breaks when it hits this:
`conn.admin_set_user_password(
            UserPoolId=user_pool_id, Username=username, Password=password, Permanent=True
        )
`
NotImplementedError: The admin_set_user_password action has not been implemented

**Comment by bblommers** [comment]
Hi @dwevans1994, that method was implemented a couple of weeks after the 1.3.16-release.
It should be supported using the latest development release, 1.3.17.dev281.
